### PR TITLE
Alerting: Add hide plugins rules filter

### PR DESCRIPTION
**Pull request by konrad147**
**What is this feature?**
This PR adds a toggle to hide plugin provided alert rules from the alert rules list

**Demo**

image

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by tomratcliffe** [comment]
At a high level - would it be better to frame this as:

>Rule origin: All, Grafana(not sure on this name), or Plugins

?

As "Exclude plugins" isn't technically an origin 🙂 And then the query param wouldn't be called "hide", but something like "origin"

File: public/app/features/alerting/unified/components/rules/RulesFilter.tsx [hunk 1]
```diff
@@ -68,9 +69,11 @@ const RuleStateOptions = Object.entries(PromAlertingRuleState).map(([key, value]
 
 const RulesFilter = ({ onFilterCleared = () => undefined }: RulesFilerProps) => {
   const styles = useStyles2(getStyles);
-  const [queryParams, setQueryParams] = useQueryParams();
+  const [queryParams, updateQueryParams] = useURLSearchParams();
+  const { pluginsFilterEnabled } = usePluginsFilterStatus();
   const { filterState, hasActiveFilters, searchQuery, setSearchQuery, updateFilters } = useRulesFilter();
 
+  // const ruleOrigin = queryParams.get('ruleOrigin');
```
Comment (tomratcliffe) [inline public/app/features/alerting/unified/components/rules/RulesFilter.tsx:76]: Commented out code here :)

Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/rules/RulesFilter.tsx:76]: should we delete this comment?

File: public/app/features/alerting/unified/hooks/useFilteredRules.ts [hunk 2]
```diff
@@ -217,6 +225,10 @@ const reduceGroups = (filterState: RulesFilter) => {
         matchesFilterFor.ruleType = true;
       }
 
+      if ('plugins' in matchesFilterFor && filterState.plugins === 'hide') {
+        matchesFilterFor.plugins = !isPluginProvidedRule(rule);
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/hooks/useFilteredRules.ts:229]: should not be the opposite? 🤔
`matchesFilterFor.plugins = isPluginProvidedRule(rule);`

**Review by soniaAguilarPeiron — APPROVED** [review]
Code LGTM!

**Review by tomratcliffe — COMMENTED** [review]
Did you work out why it's always re-fetching the endpoints when you change the plugins filter, but not any of the others? 🤔

**Comment by konrad147** [comment]
> Did you work out why it's always re-fetching the endpoints when you change the plugins filter, but not any of the others? 🤔

That happens for other filters two, but its harder to notice :)
If you select a filter we need to remove the `limitAlerts` parameter which triggers reloading all rules
If you have no other filters applied and you only switch between `Hide` and `Show` plugin rules you effectively toggle between clean all filters (Show option - limitAlerts applied) and pluging rules filter (Hide option - limitAlerts disabled)

**Review by tomratcliffe — APPROVED** [review]
LGTM!

**Pull request merged**